chain-signatures/contract-eth/contracts/ChainSignatures.sol

```diff
@@ -27,144 +25,55 @@ contract ChainSignatures is Initializable {
-    }
-
-    function verifyDerivedKey(uint256 epsilon, PublicKey memory derivedPk) private view returns (bool) {
-        (uint256 epsilonGx, uint256 epsilonGy) = Secp256k1.ecSub(derivedPk.x, derivedPk.y, publicKey.x, publicKey.y);
-        // verify epsilonGx, epsilonGy is G * epsilon, based on:
-        // https://ethresear.ch/t/you-can-kinda-abuse-ecrecover-to-do-ecmul-in-secp256k1-today/2384
-        address recovered = ecrecover(bytes32(0), 27, bytes32(Secp256k1.GX), bytes32(mulmod(Secp256k1.GX, epsilon, Secp256k1.N)));
-
-        // epsilonG should match recovered
-        bytes32 epsilonGHash = keccak256(abi.encodePacked(epsilonGx, epsilonGy));
-        address epsilonGAddr = address(uint160(uint256(epsilonGHash)));
-        return epsilonGAddr == recovered;
-    }
-
-    function deriveEpsilon(string memory path, address requester) public pure returns (uint256) {
-        string memory requesterStr = Strings.toHexString(uint256(uint160(requester)), 20);
-        string memory epsilonString = string.concat("near-mpc-recovery v0.2.0 epsilon derivation:", requesterStr, ",", path);
-        console.log("Epsilon String:", epsilonString);
-        bytes32 epsilonBytes = keccak256(bytes(epsilonString));
-        uint256 epsilon = uint256(epsilonBytes);
-        return epsilon;
-    }
-
-    function latestKeyVersion() public pure returns (uint32) {
-        return 0;
+    constructor(address _admin) {
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+        signatureDeposit = 50000 gwei;
     }
 
     function sign(SignRequest memory _request) external payable returns (bytes32) {
         bytes32 payload = _request.payload;
-        string memory path = _request.path;
+        string memory chainIdStr = Strings.toString(block.chainid);
+        string memory path = string.concat(_request.path, "/", chainIdStr);
         uint32 keyVersion = _request.keyVersion;
+        string memory algo = _request.algo;
+        string memory dest = _request.dest;
+        string memory params = _request.params;
 
-        if (keyVersion > latestKeyVersion()) {
-            revert("This key version is unsupported. Call latestKeyVersion() to get the latest supported version");
-        }
-
-        uint256 requiredDeposit = getSignatureDeposit();
-        require(msg.value >= requiredDeposit, "Insufficient deposit");
-
-        // Concert payload to int as big-endian, check if payload is than the secp256k1 curve order
-        uint256 payloadHash = uint256(payload);
-        require(
-            payloadHash < 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141,
-            "Payload exceeds secp256k1 curve order"
-        );
-
-        bytes32 requestId = keccak256(abi.encodePacked(payload, msg.sender, path));
-        require(pendingRequests[requestId].requester == address(0), "Request already exists");
+        bytes32 requestId = keccak256(abi.encodePacked(payload, msg.sender, path, keyVersion, algo, dest, params));
```

Comment:
this potentially can cause collisions very easily due to how `encodePacked` works (for example`encodePacked("a", "bc") == encodePacked("ab", "c")`, so for now, we can add in separators to make it less so and also the block timestamp:
```suggestion
        bytes32 requestId = keccak256(abi.encodePacked(payload, ":", msg.sender, ":", path, ":", keyVersion, ":", algo, ":", dest, ":", params, ":", block.timestamp));
```